### ISSUE TITLE
Links doesn't open when clicked in Linux AppImage

### ISSUE THREAD

**Issue by YoussefMo7amed** [gitbutlerapp/gitbutler#2765]
![Screenshot_20240216_234747](https://github.com/gitbutlerapp/gitbutler/assets/39943970/5d02bc7b-e52b-45d2-b30a-23861aa1c307)
![Screenshot_20240216_235113](https://github.com/gitbutlerapp/gitbutler/assets/39943970/19c07f26-1abb-4362-bc46-88edbe146ebe)

Links doesn't open when clicked
- Linux Manjaro with KDE environment plasma 5
- using `git-butler_0.10.10_amd64_80a9bb119838942fba0b9f79fa6620cc.AppImage`

**Comment by theDevelopper** [comment]
This also seems to prevent logging in

**Comment by JokerQyou** [comment]
Reproducible with Manjaro Linux (KDE plasma 6). Also the menu links does not work, clicking on `Help > Documentation` does nothing.

**Comment by ndom91** [comment]
> Hmm, I wonder if the issue could be specific to KDE. @JokerQyou which app version did you test with? Is it a recent one?
> cc @ndom91 

Interesting, I just opened a PR which should fix this (https://github.com/gitbutlerapp/gitbutler/pull/4706)

We package the xdg-open binary in with the appimage, but it doesn't seem to be able to execute it to open links. If you open the appimage from the terminal and attempt to click a link, you should see it throw a shell error from tauri.

**Comment by ndom91** [comment]
I would love to know if folks are still running into this issue with the latest releases (https://app.gitbutler.com/download).

I'm able to reproduce the broken xdg-open behavior on my local machine with NixOS. However, the AppImage open external link functionality works without issue on an Ubuntu VM of mine. 

If you could mention your distro and version that would help me narrow down the platforms a lot :pray:

**Comment by JokerQyou** [comment]
v0.12.20 still has this issue. Sorry for the late response, somehow I missed notifications from this thread. Distro is Manjaro Linux (latest) with KDE Plasma 6.0.5 (also latest in Manjaro), X11 flavor.

**Comment by plichard** [comment]
Same on v0.12.21, fedora 40, plasma 6.1.3, frameworks 6.4.0, wayland

**Comment by ndom91** [comment]
Thanks for the feedback! I'd love it if you could give this latest build a shot: https://github.com/gitbutlerapp/gitbutler/actions/runs/10600508432

The Ubuntu one should show up as an artifact to download from that GHA workflow run in a few min. It'll then also be published as the latest nightly shortly thereafter [here](https://app.gitbutler.com/downloads). 

Btw the GHA artifact labeled Ubuntu contains the deb, rpm and the appimage 🙏

**Comment by JokerQyou** [comment]
Nope, the nightly build (shows `0.5.700`) downloaded from the link above still does not work.

**Comment by plichard** [comment]
Same here, any particular RUST_LOG configuration I could run that could help?

**Comment by ndom91** [comment]
So the fix was just deployed in `0.5.700`, it's not merged to master yet and won't be in any following nightlies (yet).

I was able to confirm it works on the following machines:
- My local NixOS machine 
- Manjaro VM (fresh boot off their latest live CD)
- Ubuntu 22.04 VM (fresh boot off their latest live CD)

If you're having issues, it would help if you told us at least the following info:

- Which link exactly you tried in the app 
- Any relevant logs; If you AppImage via the terminal (i.e. `./git-butler-nightly.AppImage`) you'll see the output right in line in stdout.
- Which packaging method you used (AppImage, deb, rpm)

**Comment by ndom91** [comment]
Okay, another new build - this one worked for me on all of the above platforms and Fedora 39.

`git-butler-nightly_0.0.0_amd64.AppImage` - https://we.tl/t-Y2xR6TSdPl

**Comment by JokerQyou** [comment]
Just tested `0.5.704` downloaded from above link, and it does not work on Manjaro Linux.

**Comment by ndom91** [comment]
> Just tested `0.5.704` downloaded from above link, and it does not work on Manjaro Linux.

Can you provide some more details? Which link(s) did you click? Can you start the appimage from the terminal and let us know if there are any error messages?

**Comment by plichard** [comment]
Oh, I noticed this (eventually :see_no_evil: ):
```
kde-open: symbol lookup error: /usr/lib64/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer
```

`xdg-open` or `kde-open` works fine when run manually.

The library seems to come from `libproxy`.

```
Installed Packages
Name         : libproxy
Version      : 0.5.5
Release      : 1.fc40
Architecture : x86_64
Size         : 111 k
Source       : libproxy-0.5.5-1.fc40.src.rpm
Repository   : @System
From repo    : updates
```

**Comment by JokerQyou** [comment]
> > Just tested `0.5.704` downloaded from above link, and it does not work on Manjaro Linux.
> 
> Can you provide some more details? Which link(s) did you click? Can you start the appimage from the terminal and let us know if there are any error messages?

Sorry about that. I did not know the nightly build would produce debug log via stdout. I tried links on the `Before we begin` UI, as well as the menu items like `Documentation` and `X`. Here are the logs:

```
dpi is invalid got from xsettings(Qt/DPI/ and Xft/DPI), fallback to get dpi from QXcbScreen::logicalDpi()
Gtk-Message: 15:56:25.207: Failed to load module "xapp-gtk3-module"

** (git-butler-nightly:208208): WARNING **: 15:56:26.387: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
2024-09-02T07:56:26.447038Z  INFO crates/gitbutler-tauri/src/main.rs:98: starting app version=0.5.704 name=GitButler Nightly
Gtk-Message: 15:56:26.556: Failed to load module "xapp-gtk3-module"
2024-09-02T07:56:28.455421Z  INFO list_projects: crates/gitbutler-tauri/src/projects.rs:41: close time.busy=23.7µs time.idle=16.6µs
2024-09-02T07:56:28.467397Z  INFO get_user: crates/gitbutler-tauri/src/users.rs:10: close time.busy=55.7µs time.idle=28.9µs
2024-09-02T07:56:33.097410Z  INFO open_url: crates/gitbutler-tauri/src/open.rs:62: close time.busy=4.69ms time.idle=80.7µs url="https://docs.gitbutler.com/"
kde-open: symbol lookup error: /usr/lib/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer
2024-09-02T07:56:41.023983Z  INFO open_url: crates/gitbutler-tauri/src/open.rs:62: close time.busy=1.34ms time.idle=91.1µs url="https://discord.com/invite/MmFkmaJ42D"
kde-open: symbol lookup error: /usr/lib/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer
2024-09-02T07:56:48.977636Z  INFO open_url: crates/gitbutler-tauri/src/open.rs:62: close time.busy=1.38ms time.idle=113µs url="https://discord.gg/MmFkmaJ42D"
kde-open: symbol lookup error: /usr/lib/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer
2024-09-02T07:56:50.006822Z  INFO open_url: crates/gitbutler-tauri/src/open.rs:62: close time.busy=1.47ms time.idle=99.0µs url="https://gitbutler.com/privacy"
kde-open: symbol lookup error: /usr/lib/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer

** (git-butler-nightly:208208): WARNING **: 15:56:51.448: atk-bridge: get_device_events_reply: unknown signature
2024-09-02T07:56:54.383907Z  INFO open_url: crates/gitbutler-tauri/src/open.rs:62: close time.busy=1.23ms time.idle=80.4µs url="https://docs.gitbutler.com"
kde-open: symbol lookup error: /usr/lib/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer
2024-09-02T07:56:58.563932Z  INFO open_url: crates/gitbutler-tauri/src/open.rs:62: close time.busy=1.63ms time.idle=107µs url="https://x.com/gitbutler"
kde-open: symbol lookup error: /usr/lib/libproxy/libpxbackend-1.0.so: undefined symbol: g_once_init_leave_pointer

```

**Comment by ndom91** [comment]
Okay thanks a lot for the feedback, so you're both using KDE? Looks like there might be one env var that's still missing that needs to be cleaned up that that `libproxy` file looks up :thinking:

Btw, the work on this is being done in [this PR](https://github.com/gitbutlerapp/gitbutler/pull/4748)

**Comment by ndom91** [comment]
Ahh so in the fix PR we [removed all paths](https://github.com/gitbutlerapp/gitbutler/pull/4748/files#diff-749e0802f8a953f8fda787b1c1aa1461acfec0e38dbdc25b1bbfcdd9030db50cR24) that contain `/appimage-run`, which is where the custom libraries where mounted to on the distros I tried. I thought that was the appimage defualt.

It looks like under KDE that isn't the case and the cusotm libraries are mounted under a `/tmp/.mount_git*` directory.

![image](https://github.com/user-attachments/assets/eb249952-e175-4958-9087-6ab70e3704ca)

**Comment by ndom91** [comment]
Alright, the latest Nightly `0.5.713-1222` has antoher last tweak for KDE. Let me know if this works for you :pray: 

https://releases.gitbutler.com/releases/nightly/0.5.713-1222/linux/x86_64/git-butler-nightly_0.5.713_amd64.AppImage.tar.gz

**Comment by plichard** [comment]
It works! Thank you :tada: :tada:

**Comment by JokerQyou** [comment]
It works now.

**Comment by ndom91** [comment]
This has been fixed and merged to main last night. It's available in nightly `0.5.731+` (https://app.gitbutler.com/downloads).

Let us know if you have any issues with it :pray:

**Comment by ndom91** [comment]
I'm going to close this issue as I haven't heard back from anyone that it doesn't work for them. Of course if yuo're still having issues, please feel free to open a new issue. Make sure to include which Distro, desktop environment / window manager, etc. that yuo're using :pray:
